### PR TITLE
Migrate createPost to redux

### PR DESCRIPTION
**Pull request by enahum**
#### Summary
Affected areas:
* Post a message from Center channel
* Post a message from the RHS

#### Ticket Link
https://mattermost.atlassian.net/browse/MM-12711

#### Checklist
[Place an '[x]' (no spaces) in all applicable fields. Please remove unrelated fields.]
- [x] Ran `make check-style` to check for style errors (required for all pull requests)
- [x] Ran `make test` to ensure unit and component tests passed
- [x] Added or updated unit tests (required for all new features)

### CODE REVIEW

File: actions/post_actions.jsx [hunk 1]
```diff
@@ -116,31 +116,32 @@ export function unflagPost(postId) {
...
 }
 
-export async function createPost(post, files, success) {
-    // parse message and emit emoji event
-    const emojis = post.message.match(EMOJI_PATTERN);
-    if (emojis) {
-        for (const emoji of emojis) {
-            const trimmed = emoji.substring(1, emoji.length - 1);
-            emitEmojiPosted(trimmed);
+export function createPost(post, files) {
```
Comment (saturninoabril) [inline actions/post_actions.jsx]: Can we add test to `createPost`?

Comment (enahum) [reply actions/post_actions.jsx]: sure

File: tests/redux/actions/views/create_comment.test.jsx [hunk 2]
```diff
@@ -51,7 +51,7 @@ jest.mock('actions/global_actions.jsx', () => ({
 }));
 
 jest.mock('actions/post_actions.jsx', () => ({
-    createPost: jest.fn(),
+    createPost: jest.fn(() => ({type: 'MOCK_CREATE_POST'})),
```
[the remark this answers is not in the document]
Comment (enahum) [reply tests/redux/actions/views/create_comment.test.jsx]: the problem here is that the args are being generated in `submitPost` and is adding a pending_post_id and a timestamp that I don't really know who to mock

File: actions/post_actions.jsx [hunk 3]
```diff
@@ -116,31 +115,32 @@ export function unflagPost(postId) {
...
-            const trimmed = emoji.substring(1, emoji.length - 1);
-            dispatch(addRecentEmoji(trimmed));
+export function createPost(post, files) {
+    return async (doDispatch) => {
+        // parse message and emit emoji event
+        const emojis = post.message.match(EMOJI_PATTERN);
+        if (emojis) {
+            for (const emoji of emojis) {
+                const trimmed = emoji.substring(1, emoji.length - 1);
+                dispatch(addRecentEmoji(trimmed));
```
Comment (jwilander) [inline actions/post_actions.jsx]: This should be `doDispatch` right?

Comment (enahum) [reply actions/post_actions.jsx]: actually I'm going to get rid of the global dispatch in this file

**Pull request merged**